### DISCUSSION TITLE
Avoid style warning `Use assert_match instead of assert ...include?`

### DISCUSSION THREAD

**Discussion by alexreg** [Homebrew/discussions#3820] [Writing Formulae/Casks]
I'm getting the style warning `Use assert_match instead of assert ...include?`. However, this warning is inappropriate in this particular case, since the string I'm searching in might not be valid UTF-8. Of course, using `include?` works in either case. What's the recommended way to avoid this warning?

**Comment by p-linnane** [comment]
Can you post what you're working on so we can take a look?

**Reply by Bo98** [reply]
To add: if there's an example code of something that does not work with `assert_match` but does with `assert ...include?` then it would be useful to know as we could adjust the style warning if it is inappropriate. We've not come across anything ourselves yet.

**Reply by alexreg** [reply]
[Here](https://gist.github.com/4299b03608490ffd25ce8e88a222a2e2                                            
) you are, line 55. The problem is that the file may not be valid UTF-8.

**Reply by Bo98** [reply]
Does this work?

```rb
assert_includes shell_output("git cat-file blob #{foo_obj_id}"), "Hello Hello Hello"
```

**Reply by alexreg** [reply]
That does, thank you. Just learnt about that function!
